ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,32 @@ -(void)deleteObject{
     [self remove];
 }
 
+-(void)getExplorerUrlSuccessCallback:(void (^)(NSString*))success errorCallback:(void (^)(NSError*))error {
+    NSLog(@"%@ getExplorerUrl",self.Id);
+    NSMutableString *path = [NSMutableString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@",
+                      self.report_id];
+    if ([self.test_name isEqualToString:@"web_connectivity"])
+        [path appendString:[NSString stringWithFormat:@"&input=%@",
+                            self.url_id.url]];
+    NSURL *url = [NSURL URLWithString:path];
+    NSURLSessionDataTask *downloadTask =
+    [[NSURLSession sharedSession]
+     dataTaskWithURL:url
+     completionHandler:^(NSData *data, NSURLResponse *response, NSError *nserror) {
+         if (nserror != nil)
+             error(nserror);
+         else {
+             NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&nserror];
+             if (nserror != nil)
+                 error(nserror);
+             NSArray *resultsArray = [dic objectForKey:@"results"];
+             //TODO symbolize somehow empty array
+             if ([resultsArray count] == 0)
+                 error(nil);
```

Comment:
This code continues to be broken. One subtle detail with two callbacks (they approach that you choose) is that you should call either one or the other but never both. (With a single callback rules are simpler.)

Reply:
Same comment as above: if there is an error we need to notify the error callback, if everything is fine we trigger the success callback. 
Both can't be triggered, as long as one is triggered the function returns.

Reply:
My bad, just checked the docs and the function continues even after the cbs, weird implementation, I would have thought the opposite.
Added the returns but at this point I would prefer to use the AFNetworking class for semplicity

Reply:
> [...] weird implementation [...]

Why do you find it weird? It seems consistent to me that code inside a block behaves exactly as code outside the block. I'd have found weird the opposite, TBH.